### PR TITLE
E2E Gradle Demo

### PR DESCRIPTION
**Pull request by rohitp27**
Enhancements : 
- Adds build configuration to build the sparkdemoproject using gradle
- The e2e module now contains a demo for gradle (Follows the same steps as SBT)
- Adds a script to run this demo asking for prompts at each step
- Adds a script to clean-up the downloaded dependencies

### CODE REVIEW

File: e2e_demo/scala/cleanup.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+rm -rf sparkdemoproject-3 spark-* *.jar *.tgz *.gz hadoop-*
+rm -rf ../../pipelinecompare/warehouse/*
+rm -f /tmp/spark-migration-jars/*
+cd sparkdemoproject && gradle clean
```
Comment (holdenk) [inline e2e_demo/scala/cleanup.sh:6]: Why delete the tgz and spark dirs each run?
Also gradle clean does not make sense for sbt build

File: e2e_demo/scala/fetch_dependencies.sh [hunk 2]
```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
```
Comment (holdenk) [inline e2e_demo/scala/fetch_dependencies.sh:2]: Run with `set -e` as well

File: e2e_demo/scala/run_demo-gradle.sh [hunk 3]
```diff
@@ -29,13 +32,15 @@ SPARK3_DETAILS="spark-3.3.1-bin-hadoop2"
...
 spark_submit3="$(pwd)/${SPARK3_DETAILS}/bin/spark-submit"
 spark_sql3="$(pwd)/${SPARK3_DETAILS}/bin/spark-sql"
+prompt "Env setup done. Next we'll download dependencies."
 
 ########################################################################
 # Downloading dependencies
 ########################################################################
 
-#SKIPPING THIS PART FOR NOW. ASSUMING THIS IS RUN AFTER THE OG DEMO
+bash ./fetch_dependencies.sh $CORE_SPARK2 $SPARK2_DETAILS $SPARK3_DETAILS
```
[the remark this answers is not in the document]
Comment (holdenk) [reply e2e_demo/scala/run_demo-gradle.sh:41]: But move it to share the logic between the scripts.

**Review by holdenk — COMMENTED** [review]
I like it :) Some minor notes.

File: e2e_demo/scala/cleanup.sh (hunk 1 above)
Comment (rohitp27) [reply e2e_demo/scala/cleanup.sh:6]: I'm not sure of the best way to cleanup all the stuff in sbt. Somehow `sbt clean` still leaves the target directory intact for me. Is that supposed to happen? @holdenk 

Anyway, moving this gradle-specific thing to the other script

File: e2e_demo/scala/fetch_dependencies.sh (hunk 2 above)
Comment (rohitp27) [reply e2e_demo/scala/fetch_dependencies.sh:2]: Done

File: e2e_demo/scala/run_demo-gradle.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (rohitp27) [reply e2e_demo/scala/run_demo-gradle.sh:41]: Done. Both scripts use this shell file for setup. The logic is common now

File: e2e_demo/scala/cleanup.sh (hunk 1 above)
Comment (rohitp27) [reply e2e_demo/scala/cleanup.sh:6]: > Why delete the tgz and spark dirs each run?

Thought we anyway download it every time? So just wanted the dir to look fresh when I was tracking the execution prompt-by-prompt. Not good?

Comment (holdenk) [reply e2e_demo/scala/cleanup.sh:6]: No, the `! -f` is if file does not exist so we only download the files if they are not present (and they should not change so downloading them each time is wasteful).

File: e2e_demo/scala/run_demo-gradle.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (rohitp27) [reply e2e_demo/scala/run_demo-gradle.sh]: Pushing that

File: e2e_demo/scala/dl_dependencies.sh [hunk 4]
```diff
@@ -17,6 +17,11 @@ if [ ! -f ${SPARK3_DETAILS}.tgz ]; then
   wget https://archive.apache.org/dist/spark/spark-3.3.1/${SPARK3_DETAILS}.tgz &
 fi
 wait
+
+
+########################################################################
+# Setting variables
+########################################################################
```
Comment (holdenk) [inline e2e_demo/scala/dl_dependencies.sh:24]: This is actually just extracting the artifacts (if needed) not setting variables.

**Review by holdenk — COMMENTED** [review]
One minor comment, I love how you've added more comments to make the scripts more readable :)

File: e2e_demo/scala/dl_dependencies.sh (hunk 4 above)
Comment (rohitp27) [reply e2e_demo/scala/dl_dependencies.sh]: Brainfart. Too much copy-pasta

**Pull request merged**